In [73]:
import pandas as pd

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\Test_Cleaned\\Apple_test_cleaned.csv', encoding='latin-1')

In [ ]:
# df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3725 entries, 0 to 3724
Data columns (total 3 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   sentiment             3725 non-null   int64  
 1   sentiment:confidence  3725 non-null   float64
 2   text                  3725 non-null   object 
dtypes: float64(1), int64(1), object(1)
memory usage: 87.4+ KB


In [76]:
# df1 = df1.drop(['_golden','_unit_state','_trusted_judgments','_last_judgment_at','date','id','query','sentiment_gold'], axis=1)

In [77]:
# df1 = df1.to_csv("Apple_test_cleaned.csv", index=False)

In [78]:
# df1.isnull().sum()

##### The 82 rows with not_relevant sentiment have been removed. The DataFrame now only contains the relevant sentiment values:



In [79]:
# df1 = df1[df1['sentiment'] != 'not_relevant']
# df1['sentiment'].value_counts()

##### 79 rows with fewer than 5 words in the text column were removed (3804 → 3725 rows remaining). 

In [80]:
# print("Before:", len(df1))
# df1 = df1[df1['text'].str.split().str.len() >= 5]
# print("After:", len(df1))

In [81]:
# df1.to_csv("Apple_test_cleaned2.csv", index=False)

In [ ]:
# df1 = pd.read_csv("D:\\ML CODES\\IBM\\Test_Cleaned\\hotel_test_cleaned.csv")

In [ ]:
# df2.head()

,review,sentiment
0,One of the other reviewers has mentioned that ...,positive
1,A wonderful little production. <br /><br />The...,positive
2,I thought this was a wonderful way to spend ti...,positive
3,Basically there's a family where a little boy ...,negative
4,"Petter Mattei's ""Love in the Time of Money"" is...",positive


In [ ]:
# df2 = df2[['reviews.text']]

In [ ]:
# df2.to_csv('hotel_test_cleaned.csv',index=False)

In [ ]:
# df3 = pd.read_csv("D:\\ML CODES\\IBM\\test\\socialmedia-test.csv",encoding='latin-1')

In [243]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 515 entries, 0 to 514
Data columns (total 6 columns):
 #   Column                                                                                                           Non-Null Count  Dtype 
---  ------                                                                                                           --------------  ----- 
 0   4                                                                                                                515 non-null    int64 
 1   3                                                                                                                515 non-null    int64 
 2   Mon May 11 03:17:40 UTC 2009                                                                                     515 non-null    object
 3   kindle2                                                                                                          515 non-null    object
 4   tpryan                                       

In [ ]:
# df3 = df3[['text']]

In [263]:
df1.head()

,text
0,Reading my kindle2 Love it Lee childs is good...
1,Ok first assesment of the kindle2 it fucking r...
2,Youll love your Kindle2 Ive had mine for a fe...
3,Fair enough But i have the Kindle2 and I thi...
4,no it is too big Im quite happy with the Kindle2


In [258]:
print("df2 duplicates:", df1.duplicated().sum())


df2 duplicates: 0


In [257]:
df1.drop_duplicates(inplace=True)

In [ ]:
# df1['text']= df1[['@stellargirl I loooooooovvvvvveee my Kindle2. Not that the DX is cool, but the 2 is fantastic in its own right.']]

In [255]:
df1.isna().sum()

text    0
dtype: int64

In [ ]:
# df1 = df1.drop(['@stellargirl I loooooooovvvvvveee my Kindle2. Not that the DX is cool, but the 2 is fantastic in its own right.'],axis = 1)

##### Remove URLs

In [259]:
df1['text'] = df1['text'].str.replace(r"http\S+|www\S+", "", regex=True)

##### Remoev @ mentions


In [260]:
import re

df1["text"] = df1["text"].apply(lambda x: re.sub(r"@\w+", "", x))

##### Remove Punctuations

In [261]:
df1["text"] = df1["text"].apply(lambda x: re.sub(r"[^\w\s]", "", x))

#### handle elongated words

In [262]:


def remove_elongated_words(text):
    return re.sub(r'(.)\1{2,}', r'\1\1', text)

df1["text"] = df1["text"].apply(remove_elongated_words)

In [264]:
df1.to_csv("testdata_cleaned.csv", index=False)

In [ ]:
# df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\test.csv')

In [242]:
df1 = pd.read_csv('D:\\ML CODES\\IBM\\test\\testdata.manual.tweet.csv',encoding='latin-1')